File: frontend/wms-web/src/types/office.ts
export type OfficeType = 'Headquarters' | 'Regional' | 'Branch'

export type Office = {
  id: number
  code: string
  name: string
  city: string
  district: string | null
  type: OfficeType
  isActive: boolean
  warehouseCount: number
  createdAtUtc: string
}

export type CreateOfficeRequest = {
  code: string
  name: string
  city: string
  district: string
  type: OfficeType
}

File: backend/Wms.Api/Domain/Entities/Office.cs
using Wms.Api.Domain.Enums;

namespace Wms.Api.Domain.Entities;

public class Office
{
    public int Id { get; set; }
    public required string Code { get; set; }
    public required string Name { get; set; }
    public required string City { get; set; }
    public string? District { get; set; }
    public OfficeType Type { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public ICollection<Warehouse> Warehouses { get; set; } = [];
}
